modules/sequence/src/main/scala/lucuma/odb/sequence/data/GeneratorParams.scala

```diff
@@ -5,20 +5,71 @@ package lucuma.odb.sequence
 package data
 
 import cats.Eq
+import cats.Order.given
 import cats.data.NonEmptyList
-import cats.syntax.eq.*
+import cats.data.NonEmptyVector
 import lucuma.core.enums.CalibrationRole
 import lucuma.core.model.Target
 import lucuma.itc.client.ImagingIntegrationTimeInput
+import lucuma.itc.client.ImagingIntegrationTimeParameters
 import lucuma.itc.client.SpectroscopyIntegrationTimeInput
+import lucuma.itc.client.SpectroscopyIntegrationTimeParameters
+import lucuma.itc.client.TargetInput
 import lucuma.odb.sequence.ObservingMode
 import lucuma.odb.sequence.syntax.all.*
 import lucuma.odb.sequence.util.HashBytes
 
 import java.nio.charset.StandardCharsets.UTF_8
 
+case class GeneratorAsterismParams(
+  imaging: ImagingIntegrationTimeParameters,
+  spectroscopy: SpectroscopyIntegrationTimeParameters,
+  targets: NonEmptyList[(Target.Id, TargetInput)]
+) {
+  lazy val imagingInput: ImagingIntegrationTimeInput = 
+    ImagingIntegrationTimeInput(imaging, targets.map(_._2))
+
+  lazy val spectroscopyInput: SpectroscopyIntegrationTimeInput = 
+    SpectroscopyIntegrationTimeInput(spectroscopy, targets.map(_._2))
+
+  lazy val targetVector: NonEmptyVector[(Target.Id, TargetInput)] = targets.toNev
+}
+
+object GeneratorAsterismParams {
+  given Eq[GeneratorAsterismParams] =
+    Eq.by { a => (
+      a.imaging,
+      a.spectroscopy,
+      a.targets
+    )}
+
+  given HashBytes[ImagingIntegrationTimeParameters] = HashBytes.forJsonEncoder
```

Comment:
It is internal, yes. `HashBytes` can be used to produce an MD5 hash, since it's a byte array representation of the object, while `Hash` just produces an `Int`, therefore providing higher granularity.

Now that you mention it, maybe `boopickle` can be used instead? @swalker2m ?

Reply:
https://github.com/suzaku-io/boopickle